modules/compiler/utils/source/replace_local_module_scope_variables_pass.cpp

```diff
@@ -236,12 +236,9 @@ PreservedAnalyses compiler::utils::ReplaceLocalModuleScopeVariablesPass::run(
     // then add our element type to the struct
     structElementTypes.push_back(memberType);
 
-    // update the offset, taking into account any array length
-    unsigned int totalSize = alignment;
-    while (memberType->isArrayTy()) {
-      totalSize *= memberType->getArrayNumElements();
-      memberType = memberType->getArrayElementType();
-    }
+    // update the offset based on the type's size
+    unsigned int totalSize =
+        M.getDataLayout().getTypeAllocSize(memberType).getFixedValue();
```

Comment:
At this point, it might be worth capturing `M.getDataLayout()` somewhere before the loop. We also use it at line 206.

Also while we're here, we could maybe also add a more human-friendly assertion that the type alloc size is indeed not a scalable amount before grabbing `getFixedValue`?